common/models/user.js

```diff
@@ -596,14 +608,19 @@ module.exports = function(User) {
   };
 
   User.validatePassword = function(plain) {
-    if (typeof plain === 'string' && plain) {
+    var MAXLENGTH = 72;
+    if (typeof plain === 'string' && plain && plain.length <= MAXLENGTH) {
       return true;
     }
+    if (plain.length > MAXLENGTH) {
+      err = new Error (g.f('Password too long - maximum length is %s', MAXLEN));
+      err.statusCode = 422;
+      throw err;
+    }
```

Comment:
You should validate the other things first and reject earlier. Instead of testing the valid case above on line 612 first, you should check all the negative cases first. ie:

``
// if undefined reject
// if not a string reject
// if invalid length reject

now all valid, process accordingly